### DISCUSSION TITLE
Feature Request: Add Image and Video Support

### DISCUSSION THREAD

**Discussion by JatinVenkatesh** [apache/superset#28380] [Ideas]
I would like to request the addition of a feature that allows users to include images and videos in their dashboards and visualizations. Currently, Apache Superset offers a wide range of visualization options for exploring and analyzing data, but the ability to seamlessly integrate images and videos would greatly enhance the platform's capabilities and usability.

Feature Details:

Add Image Support: Users should be able to upload or specify the URL of an image file and display it within dashboards and visualizations. This could be useful for incorporating logos, charts, diagrams, or other visual elements alongside data visualizations.

Add Video Support: Users should be able to embed videos from popular video hosting platforms (e.g., YouTube, Vimeo) or specify the URL of a video file and display it within dashboards and visualizations. This could be valuable for including explanatory videos, tutorials, or dynamic content alongside data visualizations.

**Comment by mistercrunch** [comment]
You can do some of this through markdown in the Text "layout element" https://www.markdownguide.org/getting-started/

**Comment by tridoxx** [comment]
> I would like to request the addition of a feature that allows users to include images and videos in their dashboards and visualizations. Currently, Apache Superset offers a wide range of visualization options for exploring and analyzing data, but the ability to seamlessly integrate images and videos would greatly enhance the platform's capabilities and usability.
> 
> Feature Details:
> 
> Add Image Support: Users should be able to upload or specify the URL of an image file and display it within dashboards and visualizations. This could be useful for incorporating logos, charts, diagrams, or other visual elements alongside data visualizations.
> 
> Add Video Support: Users should be able to embed videos from popular video hosting platforms (e.g., YouTube, Vimeo) or specify the URL of a video file and display it within dashboards and visualizations. This could be valuable for including explanatory videos, tutorials, or dynamic content alongside data visualizations.

hi @JatinVenkatesh   What you can do is add iframes in the code, it is not "very safe" but really if it is only videos on YouTube and you have full control of this it should not be a problem, what you have to do is add this option HTML_SANITIZATION = False
 in the "superset_config.py", this option can be found in the file called "config.py" of the superset source code, then inside the dashboard you add a text type "layout element"  
and you are enable to use iframes
![image](https://github.com/apache/superset/assets/22879539/61a9d2d7-eda9-418c-bf10-c08a4b358333)
![image](https://github.com/apache/superset/assets/22879539/58518a38-0341-4594-b2c7-6f3d0b35c8c9)

**Comment by mistercrunch** [comment]
In theory you could create a new visualization plugin for youtube videos or web-hosted image urls and other parameters. I'd love to have a plugin system for dashboard widgets that are not visualizations too!

**Comment by mistercrunch** [comment]
Depending on your setup, you may have to alter the TALISMAN_CONFIG to allow-list certain URLs like youtube.com https://github.com/apache/superset/blob/master/superset/config.py#L1520-L1548
